packages/components/src/components/base/models/QueryInfo.ts

```diff
@@ -0,0 +1,335 @@
+    isLoading: boolean;
+    isMedia: boolean;
+    // isMetadataOverrideable: boolean;
+    // isTemporary: boolean;
+    // isUserDefined: boolean;
+    lastAction: LastActionStatus;
+    // lastUpdate: Date;
+    name: string;
+    pkCols: List<string>;
+    plural: string;
+    queryLabel: string;
+    schemaName: string;
+    schemaQuery: SchemaQuery;
+    singular: string;
+    status: QueryInfoStatus;
+    // targetContainers: List<any>;
+    title: string;
+    titleColumn: string;
+    // viewDataUrl: string;
+    views: Map<string, ViewInfo>;
+    schemaLabel: string;
+    showInsertNewButton: boolean;
+    importUrlDisabled: boolean;
+    importUrl: string;
+    insertUrlDisabled: boolean;
+    insertUrl: boolean;
+
+    static create(rawQueryInfo: any): QueryInfo {
+        let schemaQuery: SchemaQuery;
+
+        if (rawQueryInfo.schemaName && rawQueryInfo.name) {
+            schemaQuery = SchemaQuery.create(rawQueryInfo.schemaName, rawQueryInfo.name);
+        }
+
+        return new QueryInfo(Object.assign({}, rawQueryInfo, {
+            schemaQuery
+        }));
+    }
+
+    /**
+     * Use this method for creating a basic QueryInfo object with a proper schemaQuery object
+     * and columns map from a JSON object.
+     *
+     * @param queryInfoJson
+     */
+    static fromJSON(queryInfoJson: any) : QueryInfo {
+        let schemaQuery: SchemaQuery;
+
+        if (queryInfoJson.schemaName && queryInfoJson.name) {
+            schemaQuery = SchemaQuery.create(queryInfoJson.schemaName, queryInfoJson.name);
+        }
+        let columns = OrderedMap<string, QueryColumn>();
+        Object.keys(queryInfoJson.columns).forEach((columnKey) => {
+            let rawColumn = queryInfoJson.columns[columnKey];
+            columns = columns.set(rawColumn.fieldKey.toLowerCase(), QueryColumn.create(rawColumn))
+        });
+
+        return QueryInfo.create(Object.assign({}, queryInfoJson, {
+            columns,
+            schemaQuery
+        }))
+    }
+
+    constructor(values?: {[key:string]: any}) {
+        super(values);
+    }
+
+    isAppEditable(): boolean {
+        return this.appEditableTable && this.getPkCols().size > 0;
+    }
+
+    getColumn(fieldKey: string): QueryColumn {
+        if (fieldKey) {
+            return this.columns.get(fieldKey.toLowerCase());
+        }
+
+        return undefined;
+    }
+
+    isRequiredColumn(fieldKey: string): boolean {
+        const column = this.getColumn(fieldKey);
+        return column ? column.required : false;
+    }
+
+    getDisplayColumns(view?: string, omittedColumns?: List<string>): List<QueryColumn> {
+
+        if (!view) {
+            view = ViewInfo.DEFAULT_NAME;
+        }
+
+        let lowerOmit;
+        if (omittedColumns)
+            lowerOmit = toLowerSafe(omittedColumns);
+
+        const colFilter = (c) => {
+            if (lowerOmit && lowerOmit.size > 0) {
+                return c && c.fieldKey && !lowerOmit.includes(c.fieldKey.toLowerCase());
```

Comment:
c?.fieldKey

Reply:
There are several things I want to clean up in QueryInfo, but I want to defer that work, especially since this PR is going to get squash merged as we're moving the file, so the history of the changes would be completely lost. We should do it in a separate PR/branch.